docs/src/content/docs/patterns/structured-outputs.mdx

```diff
@@ -0,0 +1,155 @@
+---
+title: Structured outputs
+description: Getting structured output from LLMs with GenSX
+sidebar:
+  order: 4
+---
+
+For many workflows, you'll need to get structured output from LLMs. GenSX makes it easy to get structured, type-safe output from LLMs. This guide showcases how to use structured outputs with OpenAI models but you can follow a similar pattern with other LLM providers.
+
+All GenSX components include an input and output type. The most common use case is to use structured outputs to ensure that the response from the LLM matches the format of the component's output type.
+
+## Basic usage
+
+The most basic way to get a structured output from an LLM is to use the `response_format={{ type: "json_object" }}` option and then parse the response in the child function as shown below.
+
+First, define the input and output types for the component.
+
+```tsx
+interface ExtractEntitiesProps {
+  text: string;
+}
+
+interface ExtractEntitiesOutput {
+  people: string[];
+  places: string[];
+  organizations: string[];
+}
+```
+
+Next, define the component and use the `ChatCompletion` component to get the structured output.
+
+```tsx
+const ExtractEntities = gsx.Component<
+  ExtractEntitiesProps,
```

Comment:
Nit: the input type isn't actually defined anywhere in the example. Not sure if it needs to be, but it does mean that the example is technically not copy/pastable in current format.

Reply:
yeah, it's the same input type from the previous example. None of the code blocks are technically runnable in isolation (didn't include the basic imports). 

Riffing on the comment above, maybe the best path is to not worry about code examples being runnable in isolation in the docs but always link out to a fully runnable example.